File: luci-app-stargate/luasrc/model/cbi/stargate/client/safety.lua
local sys = require "luci.sys"
local util = require "luci.util"

m = Map("stargate", translate("Safety"))

s = m:section(NamedSection, "safety", "safety", translate("System changes"))
s.anonymous = true

backup_on_apply = s:option(Flag, "backup_on_apply", translate("Backup before apply"))
backup_on_apply.default = "1"
backup_on_apply.rmempty = false

transparent_proxy = s:option(Flag, "transparent_proxy", translate("Transparent proxy"))
transparent_proxy.default = "0"
transparent_proxy.rmempty = false

manage_firewall = s:option(Flag, "manage_firewall", translate("Manage firewall"))
manage_firewall.default = "0"
manage_firewall.rmempty = false

manage_dnsmasq = s:option(Flag, "manage_dnsmasq", translate("Manage dnsmasq"))
manage_dnsmasq.default = "0"
manage_dnsmasq.rmempty = false

diag = s:option(DummyValue, "_diagnostics", translate("Diagnostics"))
diag.rawhtml = true
function diag.cfgvalue()
  local logs = sys.exec("/usr/share/stargate/stargate.sh logs 2>/dev/null")
  if logs == "" then
    logs = translate("No logs")
  end
  return "<pre style=\"white-space: pre-wrap\">" .. util.pcdata(logs) .. "</pre>"
end

return m

File: luci-app-stargate/htdocs/luci-static/resources/view/stargate/safety.js
'use strict';
'require view';
'require form';
'require fs';
'require ui';

return view.extend({
  render: function() {
    var m = new form.Map('stargate', _('Safety'));

    var s = m.section(form.NamedSection, 'safety', 'safety', _('System changes'));
    s.anonymous = true;

    var backup = s.option(form.Flag, 'backup_on_apply', _('Backup before apply'));
    backup.default = '1';
    backup.rmempty = false;

    var tp = s.option(form.Flag, 'transparent_proxy', _('Transparent proxy'));
    tp.default = '0';
    tp.rmempty = false;

    var fw = s.option(form.Flag, 'manage_firewall', _('Manage firewall'));
    fw.default = '0';
    fw.rmempty = false;

    var dnsmasq = s.option(form.Flag, 'manage_dnsmasq', _('Manage dnsmasq'));
    dnsmasq.default = '0';
    dnsmasq.rmempty = false;

    var diag = s.option(form.DummyValue, '_diagnostics', _('Diagnostics'));
    diag.rawhtml = true;
    diag.cfgvalue = function() {
      return E('button', {
        'class': 'btn cbi-button cbi-button-neutral',
        'click': ui.createHandlerFn(this, function() {
          return fs.exec('/usr/share/stargate/stargate.sh', [ 'logs' ])
            .then(function(res) {
              ui.showModal(_('Recent logs'), [
                E('pre', { 'style': 'white-space: pre-wrap' }, res.stdout || _('No logs'))
              ]);
            })
            .catch(function(err) { ui.addNotification(null, E('p', {}, err.message), 'danger'); });
        })
      }, [ _('Show logs') ]);
    };

    return m.render();
  }
});
